File: Python/mario.py
import cs50


def main():
    n = size()
    x = n - 1
    for i in range(n):
        build(x, i+1)
        x = x - 1


def size():
    while (True):
        x = cs50.get_int("Size:")
        if (x > 0 and x < 9):
            break
        else:
            continue
    return x


def build(spaces, bricks):
    for _ in range(spaces):
        print(" ", end="")
    for _ in range(bricks):
        print("#", end="")
    print("")


main()

File: C Programs/mario.c
// Including the necessary header files
#include <cs50.h>
#include <stdio.h>

int size(void);
void build(int bricks, int spaces);

int main(void)
{
// Prompting the user to input pyramid's height
    int n = size();
    int x = n;
    for (int i = 0; i < n; i++)
    {
        build(i + 1, x - 1);
        x--;
    }
}

// Defining the size() function
int size(void)
{
    int n;
    do
    {
        n = get_int("Size:");
    }
    while (n < 1);
    return n;
}

// Defining the build() function
void build(int bricks, int spaces)
{
    for (int j = spaces; j > 0; j--)
    {
        printf(" ");
    }
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }
    printf("\n");
}
